src/generator/audit.jl

```diff
@@ -73,7 +73,7 @@ function sanity_check(dag::ExprDAG, options::Dict)
         elseif is_function(idn)
             ifile, iline, icol = get_file_line_column(idn.cursor)
             tfile, tline, tcol = get_file_line_column(tn.cursor)
-            error("sanity check failed. [REASON]: use the same name $tk for struct $(tn.cursor) at $tfile:$tline:$tcol and function $(idn.cursor) at $ifile:$iline:$icol. The C code use the same name for structs and functions, which is not a good code-style, please fix it in the upstream.")
+            @warn "sanity check failed. [REASON]: use the same name $tk for struct $(tn.cursor) at $tfile:$tline:$tcol and function $(idn.cursor) at $ifile:$iline:$icol. The C code use the same name for structs and functions, which is not a good code-style, please fix it in the upstream."
```

Comment:
I'm wondering what the generated code looks like. Is it possible to map these cases on the Julia side? Julia does not support duplicated struct/function identifiers.

Reply:
The code looks like:

```julia
struct stat
    st_dev::dev_t
    st_mode::mode_t
    st_nlink::nlink_t
    st_ino::__darwin_ino64_t
    st_uid::uid_t
    st_gid::gid_t
    st_rdev::dev_t
    st_atimespec::timespec
    st_mtimespec::timespec
    st_ctimespec::timespec
    st_birthtimespec::timespec
    st_size::off_t
    st_blocks::blkcnt_t
    st_blksize::blksize_t
    st_flags::__uint32_t
    st_gen::__uint32_t
    st_lspare::__int32_t
    st_qspare::NTuple{2, __int64_t}
end
```

```julia
function stat(arg1, arg2)
    @ccall stat(arg1::Ptr{Cchar}, arg2::Ptr{stat})::Cint
end
```

It seems that Julia does not have any problem with struct/function aliasing, e.g.:

```julia
% julia18
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.8.2 (2022-09-29)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> struct foo
       a::Int
       end

julia> foo(a,b,c) = a+b+c
foo

julia> foo(4)
foo(4)

julia> typeof(foo(4))
foo

julia> foo(1,2,3)
6
```

Reply:
That is actually an outer constructor of `foo`.

Reply:
```suggestion
            @warn "sanity check failed. [REASON]: use the same name $tk for struct $(tn.cursor) at $tfile:$tline:$tcol and function $(idn.cursor) at $ifile:$iline:$icol. The C code use the same name for structs and functions, which is not a good code-style, the behavior of the generated is undefined."
```